extensions/ql-vscode/src/log-insights/join-order.ts

```diff
@@ -454,7 +452,11 @@ class JoinOrderScanner implements EvaluationLogScanner {
 }
 
 export class JoinOrderScannerProvider implements EvaluationLogScannerProvider {
+  constructor(private readonly getThreshdold: () => number) {
+  }
+
   public createScanner(problemReporter: EvaluationLogProblemReporter): EvaluationLogScanner {
-    return new JoinOrderScanner(problemReporter, DEFAULT_WARNING_THRESHOLD);
+    const threshold = this.getThreshdold();
+    return new JoinOrderScanner(problemReporter, threshold);
```

Comment:
So, this means that once the logger is created, the threshold can't be changed for that current run. I think that's fine.